sdk/search/Azure.Search/src/autorest.md

```diff
@@ -35,11 +35,11 @@ directive:
 
         // Document operations
         "/docs/$count": $["/docs/$count"],
-        "/docs/search": $["/docs/search.post.search"],
+        "/docs/search.post.search": $["/docs/search.post.search"],
```

Comment:
Curious about these changes since I'll be updating this block to add in indexing. What didn't work the way you had it before (which we discussed, so I *thought* I understood what you were doing)?

Reply:
These are actually live endpoints.  Bruce would know more about why they have both, but I decided not change them when I realized it was intentional.

Reply:
This all comes down to OData-isms that actually help avoid common pitfalls of REST API design. We model the POST version of query APIs as OData "actions", which can have namespaces. In this case, the namespace is `search.post` and the action name is `search`. We support both the qualified and unqualified names for each action mainly for demo-ability and brevity.

People first evaluating an API tend to balk at things like `indexes('myindex')/docs/search.post.search` but are comfortable with `indexes/myindex/docs/search`. However, the issue with `indexes/myindex/docs/search` is that somebody could index a document whose key is `search`. In this case, we can disambiguate based on HTTP verb to tell whether it's a Search or a Lookup request, but in general the name-clashing problem is real. By always using the fully-qualified OData syntax in the client libraries, we avoid these problems completely.

Reply:
Okay, so this was to basically eliminate the potential for name collisions in our libraries (i.e. this exact change; not the content itself)?